go/ai/generate.go

```diff
@@ -172,6 +174,18 @@ func GenerateWithRequest(ctx context.Context, r *registry.Registry, opts *Genera
 		return nil, err
 	}
 
+	metadata, err := modelMetadata(r, opts.Model)
+	if err != nil {
+		return nil, err
+	}
+
+	switch metadata.Stage {
+	case ModelStageDeprecated:
+		slog.Warn(fmt.Sprintf("model: %s is deprecated and may be removed in a future release", opts.Model))
```

Comment:
In JS the warning flag is always set to `true`, that's why we are always logging here.

QQ: Is this the right way to log these types of warnings?